tests/test-runner.cpp

```diff
@@ -10,6 +10,20 @@
 int
 main(int argc, char ** argv)
 {
-  assert(argc == 2);
-  return jlm::tests::run_unit_test(argv[1]);
+  if (argc < 2)
+  {
+    return jlm::tests::RunAllUnitTests();
+  }
+  else
+  {
+    int fail = 0;
+    for (int n = 1; n < argc; ++n)
+    {
+      if (jlm::tests::run_unit_test(argv[n]))
+      {
+        fail = 1;
+      }
+    }
+    return fail;
+  }
```

Comment:
Well you can still run them manually and may want to execute just a single failing test case (e.g. under gdb).

Reply:
I still do not seem to fully understand. As it currently is, we create a single executable for each unit test file:

```
g++ -o build/tests/jlm/rvsdg/test-nodes build/tests/jlm/rvsdg/test-nodes.la   build/librvsdg.a  build/libutil.a  build/libjlmtest.a build/librvsdg.a  -pthread
```

(I just noticed that build/librvsdg.a is provided two times in the above line, but that is aside the point right now)

Such an executable is build by taking a single test file, linking this with libjlmtest.a, which contains the test-runner main() function (and some other dependencies) from above. Now, since we only have a single test file in this executable, the only test that gets registered is the one from the single test file. In the example above that would be `jlm/rvsdg/test-nodes`.

I can take this executable now and simply invoke to run the test as follows:

```
./test-nodes
```

This will go in the if case of the above if-statement and run all unit tests. Since only a single unit test file is registered as explained above, we would only run the `jlm/rvsdg/test-nodes` unit tests.

I could also go ahead and invoke the executable as follows:

```
./test-nodes jlm/rvsdg/test-nodes
```

This will go in the else case of the above if-statement and run this particular unit test. The effect is the same as invoking `./test-nodes` without argument. Since only a single unit test file is compiled into an executable, the two cases of the if-statement effectively are the same for every compiled unit test file executable. We currently do not build an executable that contains all unit test files, where it would make sense to be able to select a unit test. In other words, my point is that a selection mechanism where we can only select a single option feels redundant.

Maybe I am horribly misunderstanding something here?

Reply:
There are some files that register multiple tests, e.g. tests/jlm/rvsdg/test-graph.cpp. That is also the norm in e.g. gtest setups to bundle "related" tests.

Reply:
I was not aware of this. Thanks for the clarification.